code/datums/ai/sanity/_sanityloss_controller.dm

```diff
@@ -188,12 +188,7 @@
 /turf/proc/reachableTurftestWithMobs(caller, turf/T, ID, simulated_only)
 	if(T && !T.density && !(simulated_only && SSpathfinder.space_type_cache[T.type]) && !LinkBlockedWithAccess(T,caller, ID))
 		for(var/mob/living/L in T)
-			if(L.density)
-				return FALSE
-		for(var/obj/machinery/M in T)
-			if(istype(M, /obj/machinery/door))
-				continue
-			if(M.density)
+			if(!L.CanPass(caller, T))
```

Comment:
This no longer counts for machinery in the path, such as doors, like this. The same proc would work for doors as and objects in general here as well. Actually it'd work for all atoms in the turf if you wanted to check that.

Reply:
LinkBlockedWithAccess already does all the object checks